gitlab.go

```diff
@@ -716,8 +709,14 @@ func (c *Client) configureLimiter() error {
 	return nil
 }
 
+// BaseURL return a copy of the baseURL.
+func (c *Client) BaseURL() *url.URL {
```

Comment:
I wonder why not make it return a value, not a pointer? Same with the field in the `Client` struct - it could be just a value.

Reply:
Looks like there actually was a reason for it: https://github.com/xanzy/go-gitlab/commit/b8b6e4b888142f936ba215f02fe7712c56e29de0

Reply:
I see. Go disallows calling pointer methods on return values that are not pointers. But assigning to an intermediate variable and then calling methods on it should work. Not a big deal, just though that since a copy is being returned anyway, why not avoid extra pointer.

Reply:
Yeah, but it's debatable if that makes things easier or not. In the stdlib they pass URL as a pointer everywhere, so I probably (it's 5 years ago) though it would make sense to be consistent with that...